pkg/router/controller/status.go

```diff
@@ -112,17 +128,34 @@ func (a *StatusAdmitter) RecordRouteRejection(route *routev1.Route, reason, mess
 	})
 }
 
+// RecordRouteDeprecated attempts to update the route status with a reason for a route being deprecated.
+func (a *StatusAdmitter) RecordRouteDeprecated(route *routev1.Route, reason, message string) {
+	performIngressConditionUpdate("deprecated", a.lease, a.tracker, a.client, a.lister, route, a.routerName, a.routerCanonicalHostname, routev1.RouteIngressCondition{
+		Type:    routev1.RouteDeprecated,
+		Status:  corev1.ConditionTrue,
+		Reason:  reason,
+		Message: message,
+	})
+}
+
+// RecordRouteNotDeprecated clears the deprecated status back to an unset state.
+func (a *StatusAdmitter) RecordRouteNotDeprecated(route *routev1.Route) {
+	log.V(4).Info("RecordRouteNotDeprecated")
+	performIngressConditionRemoval("not-deprecated", a.lease, a.tracker, a.client, a.lister, route, a.routerName, routev1.RouteDeprecated)
+}
+
 // performIngressConditionUpdate updates the route to the appropriate status for the provided condition.
 func performIngressConditionUpdate(action string, lease writerlease.Lease, tracker ContentionTracker, oc client.RoutesGetter, lister routelisters.RouteLister, route *routev1.Route, routerName, hostName string, condition routev1.RouteIngressCondition) {
-	key := string(route.UID)
+	// Key the lease off of the route UID and the condition type, as different conditions will require separate updates.
+	key := leaseKey(route.UID, condition.Type)
```

Comment:
This is actually a work-item key, per our discussion.

Reply:
Done.